Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -1291,5 +1302,80 @@ public static Texture2D DownloadImage(string url)
 
             return result;
         }
+
+        /// <summary>
+        /// Data class returned from the CredentialProvider.VSS.exe tool. The tool returns this datatype as a
+        /// JSON encoded string which can be decoded using Unity's JsonUtility.
+        /// </summary>
+        [System.Serializable]
+        private struct VSSCredentialResponse
+        {
+            public string __VssPasswordWarning;
+            public string Username;
+            public string Password;
+        }
+
+        /// <summary>
+        /// Helper function to aquire a token to access VSTS hosted nuget feeds by using the CredentialProvider.VSS.exe
+        /// tool. Downloading it from the VSTS instance if needed.
+        /// </summary>
+        /// <param name="packageHost">The hostname where the VSTS instance is hosted (such as microsoft.pkgs.visualsudio.com</param>
+        /// <returns>The password in the form of a token, or null if the password could not be aquired</returns>
+        private static string GetPasswordFromVSTSCredentialProvider(string packageHost)
+        {
+            string credentialProviderBundleFilename = "CredentialProviderBundle.zip";
+            string credentialProviderFilename = "credentialprovider.vss.exe";
+            string credentialProviderPath = Path.Combine(NugetConfigFile.RepositoryPath, credentialProviderFilename);
+
+            // Download CredentialProvider if it's not already installed
+            if (!File.Exists(credentialProviderPath))
```

Comment:
Maybe the first check that should be done is to see if the credential provider path is found in an environment variable?